bugfix/deleteUserFromAdminPage

DotsenkoS opened the pull request:
Исправлено:
- На странице юзера неправильно открывались и не работали модалки delete и edit из таблицы юзеров  
- Не работал выбор списка всех юзеров на странице админа (show all users) 
- Для незалогиненного пользователя при переходе на любую страницу выпадала ошибка
- При входе в кабинет админа появлялась ошибка Uncaught (in promise) ReferenceError: moment is not define.

DotsenkoS committed:
Исправлено:
- На странице юзера неправильно открывались и не работали модалки delete и edit из таблицы юзеров
- Не работал выбор списка всех юзеров на странице админа (show all users)
- Для незалогиненного пользователя при переходе на любую страницу выпадала ошибка
- При входе в кабинет админа появлялась ошибка Uncaught (in promise) ReferenceError: moment is not define.

DotsenkoS committed:
Merge branch 'dev' of https://github.com/tediik/online_store into bugfix/deleteUserFromAdminPage

 Conflicts:
	src/main/java/com/jm/online_store/controller/rest/customer/CustomerRestController.java

DotsenkoS committed:
Исправлено:
- На странице юзера неправильно открывались и не работали модалки delete и edit из таблицы юзеров
- Не работал выбор списка всех юзеров на странице админа (show all users)
- Для незалогиненного пользователя при переходе на любую страницу выпадала ошибка
- При входе в кабинет админа появлялась ошибка Uncaught (in promise) ReferenceError: moment is not define.

softozhik commented on src/main/java/com/jm/online_store/controller/rest/admin/AdminRestController.java:
```diff
@@ -101,14 +101,16 @@
         try {
             userService.deleteByID(id);
         } catch (IllegalArgumentException e) {
+            e.printStackTrace();
```
stacktrace для отладки выводится, здесь он лишний, кмк

softozhik commented on src/main/java/com/jm/online_store/controller/rest/customer/CustomerRestController.java:
```diff
@@ -108,9 +108,18 @@ public ResponseEntity changePassword(Model model,
+            customerService.changeCustomerStatusToLocked(id);
+        }
+        catch (IllegalArgumentException e) {
+            e.printStackTrace();
```
тоже stacktrace не нужен

softozhik commented on src/main/resources/static/js/admin-page-render.js:
```diff
@@ -264,13 +262,28 @@ function handleAcceptButtonFromModalWindow(event) {
+                            headers: headers,
+                            method: 'DELETE'
+                        }).then(response => response.json())
+                            .then(blockedUser => console.log('Customer: ' + blockedUser.email + ' with ID: ' + blockedUser.id + '. Status = Locked'))
```
console.log тоже для отладки, здесь не нужны

softozhik commented on src/main/resources/static/js/profile-admin-manager-service.js:
```diff
@@ -91,8 +91,8 @@ function getCurrentUser() {
-            $("#date_birthday_input").val(moment(currentUser.birthdayDate).format("yyyy-MM-DD"));
-            $("#register_date").val(moment(currentUser.registerDate).format("yyyy-MM-DD"));
+            $("#date_birthday_input").val(currentUser.birthdayDate);
+            $("#register_date").val(currentUser.registerDate);
```
это все в этой версии берется через thyamleaf из html, я поправил в ветке про модераторов и в поле типа span не вставляется  через .val, как выяснилось, через .html вставляется
т.е. думаю, эти правки не нужны или при мерже принять мои :-)

softozhik commented:
если я правильно понял логику, то customer при удалении блокируется, а user удаляется? причем, при неудавшемся удалении, из списка пропадает, а при обновлении страницы снова появляется

DotsenkoS committed:
- при успешном удалении \ блокировке добавлено всплывающее окошко что все удачно
- при возникновлении ошибки удаления \ блокировки, в модальное окно добавлено соответствующее сообщение
-убраны лишние console.log и printStackTrace

DotsenkoS committed:
Merge branch 'dev' of https://github.com/tediik/online_store into bugfix/deleteUserFromAdminPage

 Conflicts:
	src/main/resources/static/js/profile-admin-manager-service.js

tediik commented on src/main/java/com/jm/online_store/controller/rest/admin/AdminRestController.java:
```diff
@@ -101,14 +101,15 @@
             @ApiResponse(code = 200, message = "User was deleted successfully"),
     })
     public ResponseEntity<User> deleteUser(@PathVariable Long id) {
+        User deletedUserInfo = new User(id, userService.findUserById(id).getUsername());
```
не понял этого "финта ушами", почему нельзя было просто:
User deletedUserInfo = userService.findUserById(id) ??

DotsenkoS replied:
User deletedUserInfo = userService.findUserById(id)  - не получится -  так deletedUserInfo будет ссылкой на юзера которого удаляем, и получится "не определен". Поэтому делаю отдельного с  необходимой информацией.

tediik commented on src/main/java/com/jm/online_store/controller/rest/customer/CustomerRestController.java:
```diff
@@ -108,9 +108,17 @@ public ResponseEntity changePassword(Model model,
+            log.debug("There is no user with id: {}", id);
+            return new ResponseEntity<>(HttpStatus.NO_CONTENT);
+        }
+        log.debug("User with id: {}, was deleted successfully", id);
```
по идее мы не удаляем и нехорошо вводить в заблуждение, надо написать, что просто блокируем

tediik commented on src/main/java/com/jm/online_store/controller/rest/customer/CustomerRestController.java:
```diff
@@ -108,9 +108,17 @@ public ResponseEntity changePassword(Model model,
-        customerService.changeCustomerStatusToLocked(id);
-        return ResponseEntity.ok("Delete profile");
+    public ResponseEntity<User> deleteProfile(@PathVariable Long id) {
+        User deletedUser = userService.findUserById(id);
```
раз мы просто блокируем, а не удаляем, давай по уму и метод переименуем и переменную например:
User lockedUser

tediik commented on src/main/java/com/jm/online_store/controller/rest/customer/CustomerRestController.java:
```diff
@@ -165,14 +173,18 @@ public ResponseEntity changePassword(Model model,
+            } else recentlyViewedProductsService.updateRecentlyViewedProducts(productId, userId, localDateTime);
+            return ResponseEntity.ok("Product is saved in session");
+        } else {
+            return  ResponseEntity.ok("User is not authenticated. Product is not saved to session");
```
2 пробела )))

DotsenkoS committed:
Скорректировано название метода, переменной и описание
deleteProfile -> blockProfile
deletedUser -> blockedUser

DotsenkoS committed:
Merge branch 'dev' of https://github.com/tediik/online_store into bugfix/deleteUserFromAdminPage

DotsenkoS committed:
При блокировке клиента и удалении юзера из ответа убран юзер, возвращаем только статус

tediik commented on src/main/resources/static/js/profile-admin-manager-service.js:
```diff
@@ -8,6 +10,27 @@ $(document).ready(function () {
     document.getElementById('updateProfile').addEventListener("click", updateProfile)
 });
 
+/**
+ * Функция удаления профиля
+ * @param event событие click по ссылке Удалить профиль
+ */
+function deleteProfile(event) {
+    let id = event.target.dataset.delId
+    fetch(`/api/profile/delete/${id}`, {
+        method: 'DELETE',
+        headers: {
+            'Accept': 'application/json, text/plain, */*',
+            'Content-type': 'application/json'
+        }
+    }).then(function (response) {
+        if (response.ok) {
+            document.location.href = "/logout";
+        } else {
+            toastr.error('Ваш профиль не был удален.', {timeOut: 3000});
+        }
+    })
+}
+
```
погоди, на одном из созвонов мы решали, что запрещаем нашим сотрудникам (все, кто не покупатель) выпиливаться из системы самостоятельно, т.е. они не могут удалять свои профили! Их может удалять только админ!

DotsenkoS committed:
Merge branch 'dev' of https://github.com/tediik/online_store into bugfix/deleteUserFromAdminPage

DotsenkoS committed:
Приведено в соответствие с dev

DotsenkoS committed:
Merge branch 'dev' into bugfix/deleteUserFromAdminPage
